code/noRAM/app/src/main/java/com/example/noram/QrScanFragment.java

```diff
@@ -140,7 +145,14 @@ private void processQRCode(String qrCodeString) {
                 QRType qrType = QRType.valueOf(qrDocument.getString("type"));
 
                 if (qrType == QRType.SIGN_IN) {
-                    EventManager.checkInToEvent(eventId);
+                    Location attendeeLocation = null;
+                    if (MainActivity.attendee.getAllowLocation()) {
+                        FusedLocationProviderClient fusedLocationClient = LocationServices.getFusedLocationProviderClient(getActivity());
+                        //TODO add a onsuccess and onfail listener
+                        //grabbing current loaction of the phone
+                        attendeeLocation = fusedLocationClient.getLastLocation().getResult();
```

Comment:
this line does not work - "Task is not yet completed" - you probably need a callback

Reply:
thx for running it cole :)